rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,159 @@
+![Condensed Column-Grid](./images/spatial/col-grid-spacious.png)
+
+These column grids can also be used internally based on the what component we are building. For example, let's say we are building inline form, then we'll use condensed col-grid for organising elements.
+
+### **Spatial Scale**
+We'll follow a **8pt** linear scale for the elements with the **4pt** half-step for spacing smaller content (icons, etc.) and typography (line-heights).
+
+This is how the scale will look like:
+
+| Alias      | Value |
+|------------|-------|
+| $spacer-0  | 0px   |
+| $spacer-1  | 4px   |
+| $spacer-2  | 8px   |
+| $spacer-3  | 12px  |
+| $spacer-4  | 16px  |
+| $spacer-5  | 20px  |
+| $spacer-6  | 24px  |
+| $spacer-7  | 32px  |
+| $spacer-8  | 40px  |
+| $spacer-9  | 48px  |
+| $spacer-10 | 56px  |
+
+>These are mixed values, it contains multiple of 4 as well as 8.
+
+# Drawbacks/Constraints
+There are no major drawbacks here,
+- We need to make sure that all the spacing between the elements should not be hardcoded. They should use the spatial tokens whenever required.
+- We might find few screens where it is difficult to organise the spacing. For example, iPhone 6 (`375x667`). For such scenarios we don't need to break the pixels, instead we need to keep the margins & paddings consistent. It's ok to have oodly-sized layout which keeps the grid consistent. Although, no user will likely see the actual measurements here.
+
+# Alternatives
+There are many other spatial systems we might have considered, such as **5pt grid**, **6pt grid**. But we are not using them because of already mentioned reason,
+- With odd numbers like 5, it is hard to align things to center as the outcome will result in fraction values that will split the pixels. This can also result in blurry pixel.
+- Similarly the scaling will also not work properly while scaling on mobile/tablets, which will again result into pixel blurriness.
+
+# Adoption strategy
+Spatial scale needs to be implemented as a core part of the DS in terms of the tokens which will automatically derive the adoption.
+- Implement the [spatial scale](#spatial-scale) (as tokens).
+- Implement the 3 types of [column-grid](#column-grid) structure using the same spatial scale.
+- Make sure all the components are using the same spatial scale tokens while building them.
+
+# How do we educate people?
+Both **designers** & **developers** need to be educated.
+- Designers should always use the spatial (multiple of 4pt) scale while working on any layout.
+- Similarly, developers should always use the spatial tokens while defining the space between the adjacent elements inside the UI which should not be a problem as it is coming from a designer.
+
+# Open Questions
+- We can name the tokens in a better way, just like how we have did for colors.
+- We already have static tokens (i.e. which works for both desktops & mobiles) in place, should be need dynamic tokens that changes with the device?
```

Comment:
By dynamic tokens does this mean a `space-4` token that could be say 16px on desktop and 12px on mobile?

Reply:
@divyanshu013 Yes! Somewhat similar to this - https://spectrum.adobe.com/page/design-tokens/#Global-tokens

Reply:
Thanks for clarifying. I think we can add few usecases for dynamic tokens to help decide on this.

Reply:
This depends on the usability testing only then we'll be able to identify if we need different scale on mobile